test/integration/smoke/test_diagnostics.py

```diff
@@ -0,0 +1,482 @@
+        cmd_response = self.apiclient.executeDiagnostics(cmd)
+
+        self.assertEqual(
+            '0',
+            cmd_response.EXITCODE,
+            'Failed to execute remote Arping in CPVM'
+        )
+
+
+    @attr(tags=["advanced", "advancedns", "ssh", "smoke"], required_hardware="true")
+    def test_07_traceroute_in_vr(self):
+        '''
+        Test Arping command execution in VR
+        '''
+
+        # Validate the following:
+        # 1. Arping command is executed remotely on VR
+
+        list_router_response = list_routers(
+            self.apiclient,
+            account=self.account.name,
+            domainid=self.account.domainid
+        )
+        self.assertEqual(
+            isinstance(list_router_response, list),
+            True,
+            "Check list response returns a valid list"
+        )
+        router = list_router_response[0]
+        self.debug('Starting the router with ID: %s' % router.id)
+
+        cmd = executeDiagnostics.executeDiagnosticsCmd()
+        cmd.id = router.id
+        cmd.ipaddress = '8.8.4.4'
+        cmd.type = 'traceroute'
+        cmd.params = "-m 10"
+        cmd_response = self.apiclient.executeDiagnostics(cmd)
+
+        self.assertEqual(
+            '0',
+            cmd_response.EXITCODE,
+            'Failed to execute remote Arping in VR')
+
+        # Validate Traceroute command execution with a non-existent/pingable IP address
+        cmd.ipaddress = '9999.99999.99999.9999'
+        cmd_response = self.apiclient.executeDiagnostics(cmd)
+
+        self.assertNotEqual(
+            '0',
+            cmd_response.EXITCODE,
+            'Check diagnostics command returns a non-zero exit code'
+        )
+
+    @attr(tags=["advanced", "advancedns", "ssh", "smoke"], required_hardware="true")
+    def test_08_traceroute_in_ssvm(self):
+        '''
+        Test Traceroute command execution in SSVM
+        '''
+
+        # Validate the following:
+        # 1. Traceroute command is executed remotely on SSVM
+
+        list_ssvm_response = list_ssvms(
+            self.apiclient,
+            systemvmtype='secondarystoragevm',
+            state='Running',
+        )
+
+        self.assertEqual(
+            isinstance(list_ssvm_response, list),
+            True,
+            'Check list response returns a valid list'
+        )
+        ssvm = list_ssvm_response[0]
+
+        self.debug('Setting up SSVM with ID %s' % ssvm.id)
+
+        cmd = executeDiagnostics.executeDiagnosticsCmd()
+        cmd.id = ssvm.id
+        cmd.ipaddress = '8.8.4.4'
+        cmd.type = 'traceroute'
+        cmd.params = '-m 10'
+        cmd_response = self.apiclient.executeDiagnostics(cmd)
+
+        self.assertEqual(
+            '0',
+            cmd_response.EXITCODE,
+            'Failed to execute remote Traceroute in SSVM'
+        )
+
+        # Validate Traceroute command execution with a non-existent/pingable IP address
+        cmd.ipaddress = '999.9999.9999.9999.9999'
```

Comment:
try to pass a valid but unreachable IPv4 ip see https://stackoverflow.com/questions/10456044/what-is-a-good-invalid-ip-address-to-use-for-unit-tests
Make changes to other related invalid ips you've used. Ideally, each + and - test should be a separate test case.

Reply:
Changes made, using IP address 192.0.2.2 for the - test case. Also separated the + and - test cases for the ping command.